### PR TITLE
[sonic-bgpcfgd] Call Python 3 version of sonic-cfggen for testing

### PR DESCRIPTION
**Pull request by jleveque**
**- Why I did it**

sonic-bgpcfgd build fails in the absence of Python 2, as it attempts to explicitly call sonic-cfggen using `/usr/bin/python2.7`. Also, it attempts to call sonic-cfggen using a local, relative path. Since the sonic-config-engine package is not installed, neither are its dependencies.

Resolves https://github.com/Azure/sonic-buildimage/pull/5847

**- How I did it**

Configure the Python 3 sonic-config-engine as a dependency of sonic-bgpcfgd, which ensures the Python 3 sonic-config-engine package and its dependencies are installed before sonic-bgpcfgd is built/tested.

**- How to verify it**

Build sonic-bgpcfgd.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by pavel-shirshov** [comment]
But currently sonic-config-engines supports only python2.
As soon as sonic-config-engine would be changed to python 3, your change should be applied.

**Comment by jleveque** [comment]
Python 3 sonic-config-engine should now be installed everywhere via these PRs:

- https://github.com/Azure/sonic-buildimage/pull/5154
- https://github.com/Azure/sonic-buildimage/pull/5656

**Comment by pavel-shirshov** [comment]
Yes. Youre right. That works for me.

File: src/sonic-bgpcfgd/tests/test_sonic-cfggen.py
Comment (pavel-shirshov) [inline src/sonic-bgpcfgd/tests/test_sonic-cfggen.py:16]: One suggestion is to completely remove  '/usr/bin/python3' from the command array.
Just cfggen would be enough

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py:16]: I considered that, but until we remove all Python 2, I want to explicitly call the Python 3 version, so that it doesn't accidentally try to run the Python 2 version.

File: src/sonic-bgpcfgd/tests/test_sonic-cfggen.py
Comment (pavel-shirshov) [inline src/sonic-bgpcfgd/tests/test_sonic-cfggen.py:17]: From here.
Popen will open your python interpreter

File: src/sonic-bgpcfgd/tests/test_sonic-cfggen.py
Comment (pavel-shirshov) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py:16]: ok. sounds good.

**Review by pavel-shirshov — APPROVED** [review]

File: src/sonic-bgpcfgd/tests/test_sonic-cfggen.py
Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @pavel-shirshov: When attempting to call a sonic-cfggen locally via a relative path, there were missing dependencies. Thus, I made further changes to add a test dependency on the sonic-config-engine package in order to ensure the package is installed in the build environment, and then let the location be resolved via the system paths.

However, there are now some test failures. Can you please investigate them?

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @tahmed-dev: I'm curious if these test failures may be related to the new sorting differences since you removed the dependency of sonic-cfggen on natsort. Maybe you can comment on this?

Comment (tahmed-dev) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @jleveque Yes, the errors are related to the removal of natsort from the cfggen. I thought I addressed those errors by make py2/py3 folder until Python 2 is completely removed.

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @tahmed-dev: Thanks for confirming! Do you have time to look into a solution for this? This PR was opened to unblock yet another PR. There is a chain of dependencies that are now unfortunately blocked on this.

Comment (tahmed-dev) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @jleveque welcome. is this package still being build for python2 as well?

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @tahmed-dev: For the time being, yes, we still need both Python 2 and 3 versions of sonic-config-engine.

Comment (tahmed-dev) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: I meant `bgpcfgd` packages as it looks as if the test is repeated/similar to those in `sonic-cfggen` package. @pavel-shirshov are those duplicate or just similar to `sonic-cfggen` tests? The do not have the py2/py3 mods and so they didn't cause issues when I converted the 'sonic-cfggen'. Now since they are using `sonic-cfggen` Python3 similar issues are introduced or so I think.

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @pavel-shirshov: Can you please answer @tahmed-dev's question above?

Comment (pavel-shirshov) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: As I understand, now sonic-cfggen is being run by python3. So dictionary order are different now. That was the reason I run sonic-cfggen in python2.
So I'd suggest first convert sonic-cfggen to python3. By convert I mean regenerate all the file with python3, check that python3 didn't change anything (not that easy to test), and then this PR will go smoothly.

Comment (lguohan) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: btw, @pavel-shirshov , running sonic-cfggen in python2 is causing bgpcfgd failing see issue #5897

Comment (pavel-shirshov) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @lguohan. https://github.com/Azure/sonic-buildimage/pull/5746 wasn't failed for 6 days and now it started?

Comment (lguohan) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: how close are we on this? does joe needs to regenerate all the test results?

Comment (pavel-shirshov) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: What is missing currently: dependencies of sonic-config-engine for python3. If you check why mellanox build is failing - because no netaddr installed for python3.

Also I think it is better to move sonic-config-engine to python3 first, Than continue working with this pr.

Comment (pavel-shirshov) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]:
```
File "/sonic/src/sonic-config-engine/sonic-cfggen", line 33, in <module>\n    import netaddr\nImportError: No module named netaddr\n
```

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: That's strange. The netaddr package should get installed along with sonic-config-engine: https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/setup.py#L11

If the Python 2 package of sonic-config-engine is installed, the Python 2 package of netaddr should also be installed; same for Python 3.

Comment (lguohan) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: what is the work amount to do that?

**Comment by pavel-shirshov** [comment]
retest broadcom please

**Comment by pavel-shirshov** [comment]
Can we convert sonic-cfggen into python3?

File: src/sonic-bgpcfgd/tests/test_sonic-cfggen.py
Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @tahmed-dev was looking into it.

Comment (tahmed-dev) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: > I considered that, but until we remove all Python 2, I want to explicitly call the Python 3 version, so that it doesn't accidentally try to run the Python 2 version.
> 
> I think we can clean it up later, once we have removed Python 2 completely.

@jleveque So to confirm, this package is being built only for Python 2.

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: No. @pavel-shirshov has [migrated](https://github.com/Azure/sonic-buildimage/pull/5746) the package to Python 3.

Comment (tahmed-dev) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @jleveque regenerate the verification data for Python 3 cfggen. I could not push to your branch and so create this [PR:5913](https://github.com/Azure/sonic-buildimage/pull/5913)

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: @tahmed-dev: I have incorporated your changes into my branch. A number of the tests are now passing, but some are still failing. It appears as though you may have used an out-of-date branch. I will try to fix.

Comment (jleveque) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: All check builds are passing now. Thanks for your help, @tahmed-dev!

**Review by pavel-shirshov — APPROVED** [review]

File: src/sonic-bgpcfgd/tests/test_sonic-cfggen.py
Comment (tahmed-dev) [reply src/sonic-bgpcfgd/tests/test_sonic-cfggen.py]: great news! Thanks @jleveque!

**Review by tahmed-dev — APPROVED** [review]

**Pull request merged**
